### PR TITLE
[PROTOCOL-520] - modify tests to run using existing deployments

### PR DESCRIPTION
**Pull request by dumebi**
- test using existing or not

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -11,7 +11,8 @@
     "compile": "COMPILING=true yarn hh compile",
     "build": "yarn compile && yarn types && yarn tsc",
     "types": "yarn typechain --target ethers-v5 --outDir types/typechain 'artifacts/!(build-info)/**/+([a-zA-Z0-9]).json'",
-    "coverage": "#yarn hh coverage --solcoverjs ./solcover.js"
+    "coverage": "yarn hh coverage --solcoverjs ./solcover.js",
```
Comment (passabilities) [inline package.json:14]: can leave for coverage PR as well

File: package.json [hunk 2]
```diff
@@ -65,6 +68,7 @@
     "prettier-plugin-solidity": "^1.0.0-beta.7",
     "solhint": "^3.3.4",
     "solhint-plugin-prettier": "^0.0.5",
+    "solidity-coverage": "^0.7.16",
```
Comment (passabilities) [inline package.json:71]: can leave for coverage PR as well

File: scripts/hh.sh [hunk 3]
```diff
@@ -296,12 +296,16 @@ then
...
+      ENV_VARS+="RUN_EXISTING=true "
+      opts=( ${opts[@]/"existing"} )
+  fi
+  # If a fork is already running, stop it
   fork stop 1>/dev/null
   slice_network verify
   try_fork "$network" latest
-
-  run test hardhat ${opts[*]}
+  run test localhost ${opts[*]}
```
Comment (passabilities) [inline scripts/hh.sh:308]: can revert changes

File: utils/hre-extensions.ts [hunk 4]
```diff
@@ -159,9 +170,27 @@ extendEnvironment((hre) => {
...
+          method: 'hardhat_impersonateAccount',
+          params: ['0xAFe87013dc96edE1E116a288D80FcaA0eFFE5fe5'],
+        })
+        return ethers.provider.getSigner(
+          '0xAFe87013dc96edE1E116a288D80FcaA0eFFE5fe5'
+        )
+      }
+      const accounts = await hre.getNamedAccounts()
+      return ethers.provider.getSigner(accounts[name])
+    }
```
Comment (passabilities) [inline utils/hre-extensions.ts:193]: I believe we should be able to get rid of this change as well honestly with the update to the `hardhat.config.ts`

**Review by passabilities — CHANGES_REQUESTED** [review]

File: .env.template [hunk 5]
```diff
@@ -23,3 +23,36 @@ CMC_KEY=add-your-alchemy-key-here
...
+# This is the port used to connect to the Ganache instance.
+GANACHE_PORT=8545
+
+# This is the network id used to connect to the Ganache instance.
+GANACHE_NETWORK=*
+
+# This is the gas price used to connect to the Ganache instance.
+GANACHE_GAS_PRICE=20
+
+DEPLOYER=0xAFe87013dc96edE1E116a288D80FcaA0eFFE5fe5
```
[the remark this answers is not in the document]
Comment (dumebi) [reply .env.template:58]: do you mean the template file? or deployer address

**Pull request closed without merging**